### PR TITLE
Fix role name on Azure Functions

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java [hunk 1]
```diff
@@ -200,7 +200,7 @@ public static void overlayEnvVars(Configuration config) throws IOException {
 
         if (isTrimEmpty(config.role.name)) {
             // only use WEBSITE_SITE_NAME as a fallback
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:202]: comment can be deleted too?

File: agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java [hunk 2]
```diff
@@ -190,24 +190,35 @@ public void shouldOverrideRoleName() throws IOException {
...
 
+    @Test
+    public void shouldOverrideRoleNameWithLowercaseWebsiteEnvVarOnAzFn() throws IOException {
+        envVars.set("FUNCTIONS_WORKER_RUNTIME", "java");
+        envVars.set("WEBSITE_SITE_NAME", "Role Name From Website Env");
+
+        Configuration configuration = loadConfiguration("applicationinsights_NoRole.json");
+        ConfigurationBuilder.overlayEnvVars(configuration);
+
+        assertEquals("role name from website env", configuration.role.name);
```
Comment (heyams) [inline agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java:219]: Look like case doesn't match.

Comment (trask) [reply agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java:219]: see the name of this test 😄

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:202]: I think the comment is still good, in other cases env var overrides the json config, but in this case it's only a fallback

**Review by heyams — APPROVED** [review]

**Pull request merged**